beacon-chain/types/crystallized_state.go

```diff
@@ -294,11 +289,19 @@ func (c *CrystallizedState) NewStateRecalculations(aState *ActiveState, slotNumb
 		TotalDeposits:              nextCycleBalance,
 	})
 
+	// Entering new dynasty transition.
+	if newCrystallizedState.isDynastyTransition(block.SlotNumber()) {
```

Comment:
There's a minor bug here. `isDynastyTransition` should use the new `last_finalized_slot` instead of the old one.

Reply:
It is thought?

```
	// Construct new crystallized state after cycle transition.
	newCrystallizedState = NewCrystallizedState(&pb.CrystallizedState{
		LastFinalizedSlot:          finalizedSlot,       //updated here, it's the new one
	})

	// Entering new dynasty transition.
	if newCrystallizedState.isDynastyTransition(block.SlotNumber()) {
            ....
	}
```

Reply:
Oh, my bad! I read this wrong. I think this would be slightly cleaner if `CrystallizedState` was instantiated just once, but I'll leave that up to you.

Reply:
I see what you mean. I'll change it tonight!